File: data/imported/knowledge/slides/net/merged/snippets/snippet_030.cs
[Fact]
    public void TestBlur()
    {
        // Blur effect persists.
        using var pres = new Presentation();
        var slide = pres.Slides[0];
        slide.Shapes!.Clear();
        var shape = slide.Shapes.AddAutoShape(ShapeType.Rectangle, 100, 100, 200, 100);
        var ef = shape.EffectFormat;
        ef.SetBlurEffect(8, true);

        using var pres2 = TestHelpers.SaveAndReopen(pres, _tempDir);
        var b2 = pres2.Slides[0].Shapes![0].EffectFormat.BlurEffect;
        b2.Should().NotBeNull("blur_effect should not be None after reload");
        b2!.Radius.Should().Be(8);
    }

File: data/imported/knowledge/slides/java/merged/snippets/snippet_030.java
@Test
    void testSoftEdge() throws IOException {
        try (var pres = new Presentation()) {
            var slide = clear(pres);
            var shape = slide.getShapes().addAutoShape(ShapeType.RECTANGLE, 100, 100, 200, 100);
            var ef = shape.getEffectFormat();
            ef.enableSoftEdgeEffect();
            ef.getSoftEdgeEffect().setRadius(10);

            try (var pres2 = saveAndReopen(pres)) {
                var se2 = pres2.getSlides().get(0).getShapes().get(0).getEffectFormat().getSoftEdgeEffect();
                assertThat(se2).as("soft_edge_effect should not be None after reload").isNotNull();
                assertThat(se2.getRadius()).isEqualTo(10);
            }
        }
    }